packages/mongo/oplog_tailing.js

```diff
@@ -255,12 +260,64 @@ _.extend(OplogHandle.prototype, {
 
   _maybeStartWorker: function () {
     var self = this;
-    if (self._workerActive)
-      return;
+    if (self._workerActive) return;
     self._workerActive = true;
+
     Meteor.defer(function () {
+      // May be called recursively in case of transactions.
+      function handleDoc(doc) {
+        let trigger = null;
+
+        if (typeof doc.ns === "string") {
+          [self._dbName, "admin"].some(name => {
+            const nameWithDot = name + ".";
+            if (doc.ns.startsWith(nameWithDot)) {
+              return trigger = {
+                collection: doc.ns.slice(nameWithDot.length),
+                dropCollection: false,
+                dropDatabase: false,
+                op: doc,
+              };
+            }
+          });
+        }
+
+        if (trigger === null) {
+          throw new Error("Unexpected ns");
+        }
+
+        // Is it a special command and the collection name is hidden
+        // somewhere in operator?
+        if (trigger.collection === "$cmd") {
+          if (doc.ns.startsWith("admin.")) {
+            // This was a successful transaction, so we need to apply
+            // the operations that were involved.
```

Comment:
Is there anything else we should check to be sure this was a transaction?

Reply:
Probably you should check if `applyOps` exists? I am not sure if just namespace is enough to check if it is a transaction?

Reply:
Do we have any contact at MongoDB to check this? I could not find any documentation/reference spec. This one is [obsolete](https://github.com/mongodb/mongo/wiki/Replication-Internals).

Reply:
Found [this spec](https://github.com/mongodb/specifications/blob/master/source/transactions/transactions.rst).